build.sh

```diff
@@ -29,12 +29,22 @@ fi
 
 if [ $(DOCKER_CLI_EXPERIMENTAL=enabled docker manifest inspect $build_image_name 2> /dev/null ; echo $?) -eq 0 ]; then
 	echo "----> Skipping build as image already exists"
-else
+	exit 0
+fi
+
+if [ ! -z $MAESTRO_BUILD_FOLDER ]; then
+	cd $MAESTRO_BUILD_FOLDER
+fi
+
+if [ $MAESTRO_BUILD_MODE == Dockerfile ]; then
```

Comment:
this check differs from the one in the `elif`...

if you have changed the directory why don't just check for the file presence as before?